src/main/java/com/csc530/sat/type/DDType.java

```diff
@@ -0,0 +1,15 @@
+package com.csc530.sat.type;
+
+/**
+ * A type that the decision diagram can reason about
+ *
+ * @param <T>
+ *            the java type that this type describes
+ */
+public interface DDType<T> {
```

Comment:
+ we don't have any types now that aren't just wrappers around base java types but this design allows us to write complicated types that store other data about the java classes they wrap